test/test_autograd.py

```diff
@@ -798,6 +800,73 @@ def test_grad(self):
                              + str(grad_out.shape) + " and output[0] has a shape of "
                              + str(grad_sum.shape) + ".")
 
+    def test_grad_to_node(self):
+        def check_matches(out, inp):
+            ref = torch.autograd.grad(out.sum(), inp)
+
+            grad_fn = inp.grad_fn if inp.grad_fn else inp.clone().grad_fn.next_functions[0][0]
+            edge = torch.autograd.graph.GradientEdge(grad_fn, inp.output_nr)
+
+            new = torch.autograd.grad(out.sum(), edge)
+            self.assertEqual(ref, new)
+
+        # We need to ensure that our main types of Node work (regular cpp Nodes,
+        # AccumulateGrad Nodes and custom Function)
```

Comment:
Maybe add testing for the multiple output case, e.g. with var_mean?
The inplace case might also be interesting to test.